ca/ca.go

```diff
@@ -159,16 +156,12 @@ func (a *Ca) PktTox509(pktCom []byte, caBytes []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	var payload struct {
-		Issuer   string `json:"iss"`
-		Audience string `json:"aud"`
-		Email    string `json:"email"`
-	}
-	if err := json.Unmarshal(pkt.Payload, &payload); err != nil {
+	payload := new(client.OidcClaims)
+	if err := json.Unmarshal(pkt.Payload, payload); err != nil {
 		return nil, err
 	}
 
-	if payload.Audience != requiredAudience {
+	if payload.Audience != a.reqAud {
```

Comment:
For github actions ID Tokens, there is no consistent audience to match against since the audience is just CIC commentment.

Instead it might be better to add a Provider in the Ca struct and call

`VerifyPKToken(ctx context.Context, pkt *pktoken.PKToken, provider OpenIdProvider)`

This would get us the same thing and GQ signatures.

Reply:
Agreed, I tried to leave `ca.go` mostly like it was before. I'll update `ca.go` with these changes.

Reply:
fixed in [82f79c7](https://github.com/openpubkey/openpubkey/pull/79/commits/82f79c7a8322a4b434db2f9ca51004387a5524e7)